test/SlowTests/Issues/RavenDB_14034.cs

```diff
@@ -0,0 +1,604 @@
+                    {
+                        using (var session = store.OpenSession(new SessionOptions
+                        {
+                            Database = db,
+                            TransactionMode = TransactionMode.ClusterWide
+                        }))
+                        {
+                            var current = key + i;
+                            keys[i] = current;
+
+                            session.Advanced.ClusterTransaction.CreateCompareExchangeValue(current, "Oren/" + i);
+
+                            session.SaveChanges();
+                        }
+                    }
+                }
+
+                // update one compare exchange value from each database
+
+                foreach (var db in new[] {store.Database, newDbName})
+                {
+                    using (var session = store.OpenSession(new SessionOptions
+                    {
+                        Database = db,
+                        TransactionMode = TransactionMode.ClusterWide
+                    }))
+                    {
+                        var cmpxchg = session.Advanced.ClusterTransaction.GetCompareExchangeValue<string>(keys[5]);
+
+                        session.Advanced.ClusterTransaction.UpdateCompareExchangeValue(
+                            new CompareExchangeValue<string>(
+                                cmpxchg.Key,
+                                cmpxchg.Index,
+                                "Rahien"));
+
+                        session.SaveChanges();
+                    }
+                }
+
+                // verify that GetLastCompareExchangeIndexForDatabase() returns the correct result 
+
+                foreach (var db in new[] {store.Database, newDbName})
+                {
+                    using (var session = store.OpenSession(new SessionOptions
+                    {
+                        Database = db,
+                        TransactionMode = TransactionMode.ClusterWide
+                    }))
+                    {
+                        var expectedIndex = session.Advanced.ClusterTransaction
+                            .GetCompareExchangeValues<string>(keys)
+                            .Max(cmpxchg => cmpxchg.Value.Index);
+
+                        var documentDatabase = GetDocumentDatabaseInstanceFor(store).Result;
+                        using (documentDatabase.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext transactionContext))
+                        using (transactionContext.OpenReadTransaction())
+                        {
+                            var lastCmpXchgIndex = documentDatabase.ServerStore.Cluster
+                                .GetLastCompareExchangeIndexForDatabase(transactionContext, db);
+
+                            Assert.Equal(expectedIndex, lastCmpXchgIndex);
+                        }
+                    }
+                }
+            }
+        }
+
```

Comment:
We need a test with multiple databases - see my comment above

Reply:
This test case has 2 databases and 10 compare exchange values on each db.
Should I test with more than 2 databases?

Reply:
I didn't notice it. That's great then.

Small note - instead of using `CreateDatabaseOperation` you can just call `using (var store = GetDocumentStore())` one more time. This way the database would be automatically deleted after the test. Right now the database remains on the server (most tests uses shared server instance)